package.json

```diff
@@ -20,27 +20,43 @@
     "url": "https://github.com/embark-framework/embark.git"
   },
   "bin": {
-    "embark": "./bin/embark"
+    "embark": "./dist/bin/embark.js"
   },
-  "directories": {
-    "lib": "./lib"
-  },
-  "main": "./lib/index.js",
+  "main": "./dist/lib/index.js",
   "scripts": {
-    "lint": "eslint bin/embark lib/ embark-ui/src",
-    "prepack": "cd embark-ui && npm install && npm run build",
+    "babel:node": "cross-env BABEL_ENV=node babel src --copy-files --extensions \".js,.ts\" --out-dir dist --source-maps",
+    "babel:node:watch": "npm run babel:node -- --verbose --watch",
+    "build": "npm-run-all build:*",
+    "build:node": "npm run babel:node",
+    "build:ui": "cd embark-ui && npm install && npm run build",
+    "type-check": "tsc",
+    "type-check:watch": "npm run type-check -- --watch",
+    "clean": "rimraf dist embark-*.tgz package embark-ui/build",
+    "lint": "npm-run-all lint:*",
+    "lint:js": "eslint babel.config.js src/bin src/lib embark-ui/src",
+    "lint:ts": "tslint -c tslint.json 'src/**/*.ts'",
     "test": "npm-run-all lint test:*",
-    "test:embark": "mocha test/ --no-timeouts --exit",
+    "test:embark": "mocha dist/test/ --no-timeouts --exit",
     "test:test_app": "cross-env DAPP=\"test_app\" npm run test_dapp",
     "test:contracts_app": "cross-env DAPP=\"contracts_app\" npm run test_dapp",
-    "test_dapp": "cross-env-shell \"cd test_apps/$DAPP && npm install && npm test\""
+    "test_dapp": "cross-env-shell \"cd test_apps/$DAPP && npm install && npm test\"",
+    "prepublishOnly": "npm install && npm-run-all build test",
```

Comment:
Do we want to lint here too?

Reply:
Test contains the lint:
`"test": "npm-run-all lint test:*",`